docs/design/ltc_disc.md

```diff
@@ -0,0 +1,134 @@
+# Design Doc: Accelerate PyTorch training vis BladeDISC and PyTorch Lazy Tensor Core
+
+BladeDISC is an end-to-end compiler that accelerates AI workload based on MLIR.
+PyTorch [Lazy Tensor Core (LTC)](https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/lazy_tensor_core/README.md)
+provides a infrastructure to connect to PyTorch to deep learning accelerators
+especial on training.  This design document aims to introduce how to extend
+PyTorch LTC to accelerate PyTorch training via BladeDISC.
+
+## Overview
+
+In PyTorch LTC world, TorchScript backend lowing Lazy Tensors into TorchScript
+and execute it with `torch::jit:GraphExecutor`.  LTC provides a way to add a
+new vender with implementing `BackendImplInterface`.  To BladeDISC, the ROI
+is too low to lowing all Lazy Tensors to BladeDISC IR, a better way is to extend
+TorchScript backend, which only optimizes a sub-graph of the whole TorchScript graph.
```

Comment:
Shall we call this approach something like "SubGraph" optimization mode to TSBackend?

Reply:
To my understanding this looks like a TS CUDA codegen service, in a similar position as nvFuser?

Reply:
Yes, I approved that. The main difference might be the region of interest and how we do the compilation, aka the difference between nvFuser and BladeDISC.

Reply:
I added an issue #246 to survey nvfuse , maybe we can update the implementation section in this design doc as the following content:

``` 
option1: xx
option2: xxx
I chose xxx because: 1. ... 2... .3...
```